app/controllers/course/controller.rb

```diff
@@ -1,6 +1,8 @@
 class Course::Controller < ApplicationController
   load_and_authorize_resource :course
 
+  before_action :add_course_breadcrumb, only: [:show, :index, :edit]
```

Comment:
Maybe `except: [:new]` would have expressed my intent better. The reason for it is that `course_path(current_course)` will fail for new courses which don't yet have ids.

I've removed the `before_action` filters and added an explicit check for presence of id.